packages/mds-agency/request-handlers.ts

```diff
@@ -414,65 +439,61 @@ export const submitVehicleTelemetry = async (req: AgencyApiRequest, res: AgencyA
       logger.info(name, 'no valid telemetry in', data.length, 'items:', body, 'failures:', fails)
       res.status(400).send({
         error: 'invalid_data',
-        error_description: 'none of the provided data was valid',
-        result: 'no valid telemetry submitted',
-        failures
+        error_description: 'None of the provided data was valid',
+        error_details: failures
       })
     }
   } catch (err) {
-    res.status(400).send({
-      error: 'invalid_data',
-      error_description: 'none of the provided data was valid',
-      result: 'no valid telemetry submitted',
-      failures: [`device_id ${data[0].device_id}: not found`]
+    res.status(500).send({
+      error: 'server_error',
+      error_description: 'None of the provided data was valid',
+      error_details: [` device_id ${data[0].device_id}: not found`]
     })
   }
 }
 
-export const registerStop = async (req: AgencyApiRequest, res: AgencyApiResponse) => {
+export const registerStop = async (req: AgencyApiRequest, res: AgencyRegisterStopResponse) => {
   const stop = req.body
 
   try {
     isValidStop(stop)
     const recorded_stop = await db.writeStop(stop)
-    return res.status(201).send(recorded_stop)
-  } catch (err) {
-    if (err instanceof NotFoundError) {
-      return res.status(404).send(err.message)
-    }
-    if (err instanceof ValidationError) {
-      return res.status(400).send({ error: err })
+    const { version } = res.locals
+    return res.status(201).send({ version, ...recorded_stop })
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return res.status(400).send({ error: 'invalid_data', error_description: 'Invalid stop data' })
     }
 
-    return res.status(500).send(new ServerError())
+    return res.status(500).send(agencyServerError)
```

Comment:
Could just use new ServerError here too I think because we don't have the same error structure